File: app/src/main/java/com/grohden/niceanimals/realm/entities/NiceAnimal.java
package com.grohden.niceanimals.realm.entities;

import io.realm.RealmObject;

public class NiceAnimal extends RealmObject {

    private String pictureUrl;
    private boolean isFavorite = false;
//    private AnimalType type; TODO: find out a patter for enums in realm

    public NiceAnimal() {

    }

    public NiceAnimal(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

    public String getPictureUrl() {
        return pictureUrl;
    }

    public void setPictureUrl(String pictureUrl) {
        this.pictureUrl = pictureUrl;
    }

    public boolean isFavorite() {
        return isFavorite;
    }

    public void setFavorite(boolean favorite) {
        isFavorite = favorite;
    }
}

File: app/src/main/java/com/grohden/niceanimals/realm/entities/NiceAnimal.kt
package com.grohden.niceanimals.realm.entities

import com.grohden.niceanimals.shibe.service.AnimalType
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

@RealmClass
open class NiceAnimal : RealmObject {

    @PrimaryKey
    lateinit var pictureUrl: String
    lateinit var type: String

    constructor() {

    }

    constructor(pictureUrl: String, animalType: AnimalType) {
        this.pictureUrl = pictureUrl
        this.type = animalType.name
    }
}
